engine/api/src/main/scala/pl/touk/nussknacker/engine/api/util/ReflectUtils.scala

```diff
@@ -4,7 +4,9 @@ object ReflectUtils {
 
   def fixedClassSimpleNameWithoutParentModule(clazz: Class[_]): String = {
     // fix for https://issues.scala-lang.org/browse/SI-8110
-    clazz.getName
+
+    val className = if(clazz.isArray) clazz.getCanonicalName else clazz.getName
+    className
```

Comment:
could we change all to canonical name? from what i know it would change $ sign to dot in nested classes and only would not work with anonymous inner classes

Reply:
could we just use getSimpleName ?, it look as if this issue mentioned above is long gone, unfortunately we don't have unit tests for this util method

Reply:
if it's as you say and if it's enough information for us in case of e.g. nested classes than I think we could 
I can add the tests

Reply:
Yep, after upgrade to scala 2.12 we can remove this helper. How `TypedClass` that is array of Strings will be present now? Am I right that it will be presented as a `Object[][String]` IMO it should be rather `Array[String]`. Other option is to present as a `String[]` but I think that sticking with Java/C notation is not a benefit for us - let's make this types more human readable. Can u write a test for that?

Reply:
added the tests for display, please take a look if that's what you meant